UnitystationLauncher/Models/Server.cs

```diff
@@ -34,9 +34,32 @@ public Server(string forkName, int buildVersion, string serverIp, int serverPort
             RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ? LinuxDownload :
             throw new InvalidOperationException("Failed to detect OS");
 
+        public bool HasTrustedUrlSource
+        {
+            get
+            {
+                const string trustedHost = "unitystationfile.b-cdn.net";
+                var urls = new[] { WinDownload, OsxDownload, LinuxDownload };
+                foreach (var url in urls)
+                {
+                    if (string.IsNullOrEmpty(url))
+                    {
+                        return false;
+                    }
+                    var uri = new Uri(url);
+                    if (uri.Scheme != "https" && uri.Host != trustedHost)
```

Comment:
This is `&&` when it should be `||` this allows you to use any host as long as you don't use `http` or you can use `http` and the legit host.
```suggestion
                    if (uri.Scheme != "https" || uri.Host != trustedHost)
```